src/upchunk.ts

```diff
@@ -156,11 +170,37 @@ export class UpChunk  {
     if (
       this.chunkSize &&
       (typeof this.chunkSize !== 'number' ||
-        this.chunkSize <= 0 ||
-        this.chunkSize % 256 !== 0)
+        this.chunkSize < 256 ||
+        this.chunkSize % 256 !== 0 ||
+        this.chunkSize < this.minChunkSize ||
+        this.chunkSize > this.maxChunkSize)
+    ) {
+      throw new TypeError(
+        `chunkSize must be a positive number in multiples of 256, between ${this.minChunkSize} and ${this.maxChunkSize}`
+      );
+    }
+    if (
+      this.maxChunkSize &&
+      (typeof this.maxChunkSize !== 'number' ||
+        this.maxChunkSize < 256 ||
+        this.maxChunkSize % 256 !== 0 ||
+        this.maxChunkSize < this.chunkSize ||
+        this.maxChunkSize < this.minChunkSize)
+    ) {
+      throw new TypeError(
+        `maxChunkSize must be a positive number in multiples of 256, and larger than or equal to both ${this.minChunkSize} and ${this.chunkSize}`
+      );
+    }
+    if (
+      this.minChunkSize &&
+      (typeof this.minChunkSize !== 'number' ||
+        this.minChunkSize < 256 ||
+        this.minChunkSize % 256 !== 0 ||
+        this.minChunkSize > this.chunkSize ||
```

Comment:
Should this be >= (and <= for the corresponding check on maxChunkSize)?

Reply:
good catch!

Reply:
Actually, on second thought, I think my logic here is correct.  I'm fine with minChunkSize being the same as chunkSize, and with maxChunkSize being the same as chunkSize.  In fact, in the pathological case, I'm fine with all three being exactly the same value -- it should be just exactly the same as not using dynamic chunk sizes.

I just tested this, and it worked as expected with all three being set to the same value.

Reply:
I agree. I had misread the code and thought you _should_ allow having them equal. But you already do!